### PR TITLE
Savepoint deserialization fixup - The class is an inner class, but not statically accessible.

### PR DESCRIPTION
**Pull request by raphaelsolarski**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

## Summary by CodeRabbit

- **New Features**
	- Streamlined handling of type information for Java and Scala maps, improving performance and simplifying the implementation.
	- Enhanced Component API to access information about expression parts used in SpEL templates.
	- Kafka sources and sinks can now operate with schemaless topics.
	- Lifted `TypingResult` information for dictionaries.
	- Improved handling of missing Flink Kafka Source/Sink `TypeInformation`.
	- Fixed deployment issues related to scenarios with dictionary editors after model reload.

- **Bug Fixes**
	- Addressed issues with savepoint deserialization, particularly for scenarios involving asynchronous enrichers.

- **Tests**
	- Simplified test methods to improve readability while maintaining thorough validation of serialization processes.

### CODE REVIEW

File: engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/typeinformation/internal/typedobject/TypedObjectBasedTypeInformation.scala [hunk 1]
```diff
@@ -237,7 +224,33 @@ abstract class TypedObjectBasedSerializerSnapshot[T] extends TypeSerializerSnaps
     }
   }
 
-  val buildIntermediateSchemaCompatibilityResult: BuildIntermediateSchemaCompatibilityResult
+  private def constructIntermediateCompatibilityResultProxied(
+      newNestedSerializers: Array[TypeSerializer[_]],
+      nestedSerializerSnapshots: Array[TypeSerializerSnapshot[_]]
+  ): IntermediateCompatibilityResult[_] = {
+    // signature of CompositeTypeSerializerUtil.constructIntermediateCompatibilityResult changed between flink 1.18/1.19
```
Comment (lciolecki) [inline engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/typeinformation/internal/typedobject/TypedObjectBasedTypeInformation.scala:231]: has been changed

**Review by arkadius — APPROVED** [review]
LGTM, but before merge let's check if this approach works on 1.18 as well

File: docs/Changelog.md [hunk 2]
```diff
@@ -21,6 +21,7 @@
   * [#7259](https://github.com/TouK/nussknacker/pull/7259) `flink-executor` and `lite-runtime` modules: Added compile-time
     dependency to `http-utils` (which depends on `async-http-client-backend-future` and indirectly on Netty)
 * [#7066](https://github.com/TouK/nussknacker/pull/7066) Kafka source and sink can now operate with schemaless topics. They accept any json. Data will not be validated with schema.
+* [#7270](https://github.com/TouK/nussknacker/pull/7270) Savepoint deserialization fixup - some taken savepoints (e.g. for scenarios with async enrichers) were not deserializable which led to errors during redeployments on Flink
```
Comment (arkadius) [inline docs/Changelog.md:24]: Please move the entry to 1.18.1 section

Comment (raphaelsolarski) [reply docs/Changelog.md]: moved

**Pull request merged**